integrations/retail/nemlig.js

```diff
@@ -0,0 +1,137 @@
+import moment from 'moment';
+import request from 'superagent';
+import { CookieAccessInfo } from 'cookiejar';
+
+import { ACTIVITY_TYPE_MEAL } from '../../definitions';
+
+
+/*
+Potential improvements:
+- only refetch items since last fetch.
+*/
+
+// Urls for requests
+const LOGIN_URL = 'https://www.nemlig.com/webapi/login/login';
+const ORDERS_URL = 'https://www.nemlig.com/webapi/order/GetBasicOrderHistory';
+const ORDERS_DETAILS_URL = 'https://www.nemlig.com/webapi/order/GetOrderHistory';
+
+
+// Create an agent that can hold cookies
+const agent = request.agent();
+
+/*
+  Remove cookie injected by Imperva Incapsula that contains character '\u0001'
+  which causes issues on following requests.
+*/
+function removeInvalidCookies() {
+  /*
+      Modifying cookies of a superagent agent is neither documented or supported
+      and this should be seen as a 'hack'
+  */
+  const cookies = agent.jar.getCookies(CookieAccessInfo('nemlig.com', '/'));
+  const invalidcookie = cookies.find(cookie => cookie.value.includes('\u0001'));
+  if (invalidcookie) invalidcookie.value = '';
+}
+
+// Login
+async function logIn(username, password, logger) {
+  removeInvalidCookies();
+  const res = await agent.post(LOGIN_URL).send({
+    AppInstalled: false,
+    AutoLogin: false,
+    CheckForExistingProducts: true,
+    DoMerge: true,
+    Username: username,
+    Password: password,
+  }).catch((err) => {
+    if (err.response && err.response.body.ErrorMessage) {
+      // Most likely wrong username or password
+      throw Error(err.response.body.ErrorMessage);
+    } else {
+      throw err;
+    }
+  });
+
+  logger.logDebug('Successfully logged in.');
+}
+
+async function getOrders(logger) {
+  removeInvalidCookies();
+
+  const res = await agent.get(ORDERS_URL).query({ skip: 0, take: 100 });
+
+  const json = JSON.parse(res.text);
+  logger.logDebug(`Found ${json.Orders.length} orders`);
+
+  return json.Orders;
+}
+
+async function getOrderDetails(ordernumber, logger) {
+  removeInvalidCookies();
+  const res = await agent.get(ORDERS_DETAILS_URL).query({ orderNumber: ordernumber });
+
+  const json = JSON.parse(res.text);
+  logger.logDebug(`Receipt ${ordernumber} has ${json.Lines.length} items`);
+  const deliveryDate = moment(json.DeliveryDate, 'DD/MM-YYYY').toDate();
+
+  return json.Lines.map(line => ({
+    datetime: deliveryDate,
+    activityType: ACTIVITY_TYPE_MEAL,
+    product: line.ProductName,
```

Comment:
those are the lines that would need to be modified in order to create proper activities.